File: src/Payments.Shared/Models/PaymentIntent.cs
namespace Payments.Shared.Models;

public class PaymentIntent
{
    public string Id { get; set; } = string.Empty;
    public PaymentIntentStatus Status { get; set; }
    public decimal Amount { get; set; }
    public string Currency { get; set; } = string.Empty;
    public string? Description { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public DateTime? ConfirmedAt { get; set; }
    public DateTime? ExpiresAt { get; set; }
    public DateTime? CapturedAt { get; set; }
    public DateTime? ReversedAt { get; set; }
    public DateTime? ExpiredAt { get; set; }
}

File: frontend/payments-web/types/payment-intent.ts
export type PaymentIntentStatus =
  | "Created"
  | "PendingConfirmation"
  | "Captured"
  | "Reversed"
  | "Expired";

export type PaymentIntent = {
  id: string;
  amount: number;
  currency: string;
  description?: string | null;

  status: PaymentIntentStatus;

  createdAt: string; // ISO
  confirmedAt?: string | null; // ISO
  capturedAt?: string | null; // ISO
  reversedAt?: string | null; // ISO
  expiredAt?: string | null; // ISO
  expiresAt?: string | null; // ISO (deadline de expiración)
};

export type CreatePaymentIntentRequest = {
  amount: number;
  currency: string;
  description?: string;
};
